src/nebula-hooks/use-sorting.ts

```diff
@@ -23,8 +28,15 @@ export const sortingFactory = (dimensionsLength: number, model: EngineAPI.IGener
         qOp: "Replace",
         qValue: `[${sortOrder.join(",")}]`,
       });
-    } else {
-      // Reverse
+    }
+
+    // Reverse
+    if (
+      (isNewHeadCellMenuEnabled &&
+        ((newSortDirection && newSortDirection !== sortDirection) || (!newSortDirection && colIdx === topSortIdx))) ||
+      // if flag is not enabled -> we only want to reverse if colIdx is eql to topSortedIdx
+      (!isNewHeadCellMenuEnabled && colIdx === topSortIdx)
+    ) {
```

Comment:
I agree with codeclimate here. I'd start with creating the variable `isActivelySorted` in here, and use it instead of `colIdx === topSortIdx`, since you check that three times. Then I would probably break out this into two constants, maybe one for when the flag is on and one when the flag is off

Reply:
fixed